centrallix/objectsystem/obj_datatypes.c

```diff
@@ -840,7 +853,7 @@ objDataToDouble(int data_type, void* data_ptr)
 	    {
 	    case DATA_T_INTEGER: v = *(int*)data_ptr; break;
 	    case DATA_T_STRING: v = strtod((char*)data_ptr, NULL); break;
-	    case DATA_T_MONEY: m = (pMoneyType)data_ptr; v = m->WholePart + (m->FractionPart/10000.0); break;
+	    case DATA_T_MONEY: m = (pMoneyType)data_ptr; v = m->MoneyValue/10000 + ((m->MoneyValue%10000)/10000.0); break;
```

Comment:
If you can do double arithmetic, you can just divide MoneyValue by 10000 without doing any other arithmetic